_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,315 @@
+
+```sh
+kubectl cp ca.p12 <interactive_pod_name>:/tmp
+```
+
+We can now proceed to fetch the User certificate and password for the user. User certificate and password are extracted from the User Secret like this :
+
+```sh
+kubectl get secret <kafka_user> -o jsonpath='{.data.user\.p12}' | base64 -d > user.p12
+```
+
+```sh
+kubectl get secret <kafka_user> -o jsonpath='{.data.user\.password}' | base64 -d > user.password
+```
+
+Copy the User CA certificates to the interactive pod container.
+
+```sh
+kubectl cp user.p12 <interactive_pod_name>:/tmp
+```
+
+The interactive pod now has both user and cluster CA certificates. 
+
+Next, we will configure the Kafka clients to connect to our Kafka cluster and authenticate.
+The truststore and the keystore location are mapped to their respective location (which is the folder inside the interactive pod container, since all of our commands will be executed inside the shell process of the interactive pod).
+And the configuration provider will use these values.
+We will also configure the `bootstrap.servers` option to make it easier to change address to connect to the Kafka cluster.
+
+```properties
+bootstrap.servers=my-cluster-kafka-bootstrap:9093
+security.protocol=SSL
+ssl.truststore.location=/tmp/ca.p12
+ssl.truststore.password=<truststore_password_present_in_ca_password_file>
+ssl.keystore.location=/tmp/user.p12
+ssl.keystore.password=<keystore_password_present_in_user_password_file>
+```
+
+Now we will copy the `config.properties` file to the interactive pod container:
+
+```sh
+kubectl cp config.properties <interactive_pod_name>:/tmp/config.properties
+```
+
+Lets create a `topics.json` file now. As we discussed above this file will have the topics that we need to reassign:
+
+```json
+{
+  "version": 1,
+  "topics": [
+    { "topic": "my-topic-1"},
+    { "topic": "my-topic-2"}
+  ]
+}
+```
+
+After the creation of this file, copy it to the interactive pod container since we will be running all of our commands from there:
+
+```sh
+kubectl cp topics.json <interactive_pod_name>:/tmp/topics.json
+```
+
+We can now start a shell process in our interactive pod container and run the command inside it to generate our proposal `reassignment.json` data.
+Let us start the shell process:
+
+```sh
+kubectl exec -ti <interactive_pod_name> /bin/bash
+```
+
+Now it's time for generating the `reassignment.json` file. 
+We will use the `kafka-reassign-partitions.sh` command to generate our proposal `reassignment.json` data:
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server my-cluster-kafka-bootstrap:9093 \
+--command-config /tmp/config.properties \
+--topics-to-move-json-file /tmp/topics.json \
+--broker-list 0,1,2 \
+--generate
+```
+
+Once you run this command, you will be able to see the JSON data which is generated by the partition reassignment tool.
+Now you can copy and save this data to use in the `reassignment.json` file. You can also create/alter the `reassignment.json` file
```

Comment:
Wouldn't it be best here to do something like `... > reassingment.json`? Or does that not work? Readers might not know how to get the JSON into the file.